File: DockIn.Web/src/app/Dto/ArtigoDto.ts
export interface ArtigoDto {
  artigoId: number;
  description: string;
  peso: number;
  dimensoes: number;
  classeArtigo: ArtigoClasses; // ou string/number se não tiveres enum
  createdAt: string; // ou Date
}
export enum ArtigoClasses {
  A,
  B,
  C,
}

File: DockIn.Android/app/src/main/java/pt/ipp/estg/dockin/data/model/ArtigoDto.kt
package pt.ipp.estg.dockin.data.model

import com.google.gson.annotations.SerializedName

data class ArtigoDto(
    @SerializedName("artigoId")
    val artigoId: Int,

    @SerializedName("description")
    val description: String,

    @SerializedName("peso")
    val peso: Double,

    @SerializedName("dimensoes")
    val dimensoes: Double,

    @SerializedName("classeArtigo")
    val classeArtigo: Int,

    @SerializedName("createdAt")
    val createdAt: String? = null
)

data class CriarArtigoRequest(
    @SerializedName("description")
    val description: String,

    @SerializedName("peso")
    val peso: Double,

    @SerializedName("dimensoes")
    val dimensoes: Double,

    @SerializedName("classeArtigo")
    val classeArtigo: Int
)
